### PR TITLE
WIFI-1063 Repair VIF configuration

### PR DESCRIPTION
**Pull request by RickSommerville**
Signed-off-by: Rick Sommerville <rick.sommerville@netexperience.com>

### CODE REVIEW

**Comment by RickSommerville** [comment]
This is urgent because the problems addressed are blocking many automated tests.

**Comment by RickSommerville** [comment]
I left some dead code in the RRM manager.  I opened WIFI-1144 to clean this up.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/rrm_config.c [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+		blobmsg_add_u32(b, "rssi_ignore_probe_request", conf.probe_resp_threshold);
+		blobmsg_add_u32(b, "signal_connect", conf.client_disconnect_threshold);
+		blobmsg_add_u32(b, "signal_stay", conf.client_disconnect_threshold);
+		blobmsg_add_u32(b, "bcn_rate", conf.beacon_rate);
+		blobmsg_add_u32(b, "mcast_rate", conf.mcast_rate);
+	}
+	return;
+}
+
+static bool rrm_config_update( struct schema_Wifi_RRM_Config *conf, bool addNotDelete)
```
Comment (yashvardhan-netex) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/rrm_config.c:56]: addNotDelete not used anywhere. Can we get rid of that?

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/uci.c [hunk 2]
```diff
@@ -162,12 +162,15 @@ int blob_to_uci_section(struct uci_context *uci,
 	ret = uci_lookup_ptr(uci, &p, tuple, true);
 	if (ret != UCI_OK)
 		goto err_out;
+	if (!a && !(p.flags | UCI_LOOKUP_COMPLETE))
```
Comment (yashvardhan-netex) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/uci.c:165]: Why was this needed?

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c [hunk 3]
```diff
@@ -844,32 +856,36 @@ static void hs20_vif_config(struct blob_buf *b,
...
+		if (strcmp(s->type, "wifi-iface")) continue;
 
 		ifname = uci_lookup_option_string( uci, s, "ifname" );
 		if (!strcmp(ifname,vconf->if_name)) {
 			uci_section_del(uci, "vif", "wireless", (char *)s->e.name, "wifi-iface");
 			break;
 		}
 	}
-	uci_commit_all(uci);
+	uci_commit(uci, &wireless, false);
```
Comment (yashvardhan-netex) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:883]: Is this needed considering we are setting reload_config = 1 in the next step which will commit all UCIs?

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/rrm_config.c [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+	memset(&changed, 0, sizeof(changed));
+	for (i = 0; i < rconf.vif_configs_len; i++) {
+		if (!(where = ovsdb_where_uuid("_uuid", rconf.vif_configs[i].uuid)))
+			continue;
+
+		memset(&vconf, 0, sizeof(vconf));
+		if (ovsdb_table_select_one_where(&table_Wifi_VIF_Config, where, &vconf))
+		{
+			LOG(INFO, "RRM band %s updates vif %s", conf->freq_band, vconf.if_name);
```
Comment (yashvardhan-netex) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/rrm_config.c:79]: Nitpick: May be set to DEBUG. Syslog is already quite full of opensync logs. But this can definitely wait considering the criticality of the patch.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio.c [hunk 5]
```diff
@@ -24,6 +24,8 @@
 #include "uci.h"
 #include "utils.h"
 #include "captive.h"
+#include "rrm_config.h"
+#include "vlan.h"
```
Comment (yashvardhan-netex) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio.c:28]: Looks like not needed

**Review by yashvardhan-netex — APPROVED** [review]
Overall the patch looks good.

**Comment by blogic** [comment]
merged into pending, Thanks !

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/rrm_config.c (hunk 1 above)
Comment (RickSommerville) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/rrm_config.c:56]: Yes we should.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/uci.c (hunk 2 above)
Comment (RickSommerville) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/uci.c:165]: The next line will create a section if none exists.  This line prevents a caller who only wanted to delete options from a section from creating an empty section.  For instance, this may occur if Wifi_RRM_Config is removed for a radio and a Wifi_VIF_Config is removed at the same time.  Depending on the timing,  Wifi_RRM_Config may try to delete the RRM config from a no longer existent VIF.  Without this check, we may end up with a partial wifi-iface section in wireless and confuse hostApd.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c (hunk 3 above)
Comment (RickSommerville) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:883]: It is required because this copy of wireless is local.   Effectively, the reload_config = 1, just tells the WM periodic_task() to call system("reload_config") to notify the rest of the world.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/rrm_config.c (hunk 4 above)
Comment (RickSommerville) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/rrm_config.c:79]: Should do.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio.c (hunk 5 above)
Comment (RickSommerville) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio.c:28]: Should do.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/uci.c (hunk 2 above)
Comment (yashvardhan-netex) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/uci.c:165]: Got it. Thanks!

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c (hunk 3 above)
Comment (yashvardhan-netex) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c:883]: Yeah I see.
